common/src/chain/pos.rs

```diff
@@ -24,3 +29,56 @@ pub type PoolId = Id<Pool>;
 #[derive(Eq, PartialEq, TypeName)]
 pub enum Delegation {}
 pub type DelegationId = Id<Delegation>;
+
+#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd)]
+pub struct PoSChainConfig {
+    /// The lowest possible difficulty
+    target_limit: Uint256,
+    /// Time interval between the blocks targeted by the difficulty adjustment algorithm
+    target_block_time: Duration,
+    /// The distance required to pass to allow spending the block reward
+    reward_maturity_distance: BlockDistance,
+    /// Max number of blocks required to calculate average block time. Min is 2
+    blocks_count_to_average: usize,
+}
+
+impl PoSChainConfig {
+    pub fn new(
+        target_limit: Uint256,
+        target_block_time: Duration,
+        reward_maturity_distance: BlockDistance,
+        blocks_count_to_average: usize,
+    ) -> Self {
+        Self {
+            target_limit,
+            target_block_time,
+            reward_maturity_distance,
+            blocks_count_to_average,
+        }
+    }
+
+    pub fn target_limit(&self) -> Uint256 {
+        self.target_limit
+    }
+
+    pub fn target_block_time(&self) -> Duration {
+        self.target_block_time
+    }
+
+    pub fn reward_maturity_distance(&self) -> BlockDistance {
+        self.reward_maturity_distance
+    }
+
+    pub fn blocks_count_to_average(&self) -> usize {
```

Comment:
Maybe `block_count_to_average_for_blocktime`? It's kind of unclear here.

Same for member variable.